frontend/src/app/feature/customerregistry/customer/customer-info.component.ts

```diff
@@ -120,6 +133,22 @@ export class CustomerInfoComponent implements OnInit, OnDestroy {
     }
   }
 
+  checkForRestrictedEdit(): void {
+    // ALLU-19 restrict usage to admin and invoicing roles when sap number exists
+    this.form.controls.sapCustomerNumber.valueChanges.subscribe(value => {
```

Comment:
Tästä tuli mieleen että toteutuksen voisi ehkä tehdä myös näin ilman kahta subscribea (sapCustomerNumber ei muutu?):

`    
    const sapNumber = this.form.get('sapCustomerNumber')?.value;
    const userHasRole = await this.currentUser.hasRole([RoleType.ROLE_INVOICING, RoleType.ROLE_ADMIN].map(role => RoleType[role])).toPromise();

    if (sapNumber && userHasRole) {
      this.form.enable();
    } else {
      this.form.disable();
    }
 `
    
Toiminee alkuperäiselläkin ehdotuksella mutta jos mennään sillä niin subscribejen lisäksi olisi hyvä lisätä pipe(takeUntil(this.destroy$)) tai take(1).